dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -224,11 +259,34 @@ class AppEngineCocoonService implements CocoonService {
     return statuses;
   }
 
+  List<CommitTasksStatus> _commitStatusesFromJsonFirestore(List<dynamic>? jsonCommitStatuses) {
+    assert(jsonCommitStatuses != null);
+    // TODO(chillers): Remove adapter code to just use proto fromJson method. https://github.com/flutter/cocoon/issues/441
+
+    final List<CommitTasksStatus> statuses = <CommitTasksStatus>[];
+    for (final Map<String, dynamic> jsonCommitStatus in jsonCommitStatuses!) {
+      final Map<String, dynamic> jsonCommit = jsonCommitStatus['Commit'];
+
+      statuses.add(
+        CommitTasksStatus()
+          ..commit = _commitFromJsonFirestore(jsonCommit)
+          ..branch = _branchFromJsonFirestore(jsonCommit)!
+          ..tasks.addAll(_tasksFromJsonFirestore(jsonCommitStatus['Tasks'])),
+      );
+    }
+
+    return statuses;
+  }
+
   String? _branchFromJson(Map<String, dynamic> jsonChecklist) {
     final Map<String, dynamic> checklist = jsonChecklist['Checklist'];
     return checklist['Branch'] as String?;
   }
 
+  String? _branchFromJsonFirestore(Map<String, dynamic> jsonCommit) {
+    return jsonCommit['Branch'] as String?;
```

Comment:
Consider making constants for these map keys like 'Branch", "Commit" etc.

Reply:
Done for both commit and task.